### ISSUE TITLE
Flowcept doesn't work if root-level keys are not informed in the settings file

### ISSUE THREAD

**Issue by renan-souza** [ORNL/flowcept#161]
Requirement: Flowcept should work with the default settings in the following situations:

- [ ] (1) The user specified a settings file (in the env var) that is completely empty.
- [ ] (2) The user specified a settings file (in the env var) that is partially filled.
- [ ] (3) The user did not specify any setting file. The setting file simply does not exist.

When we achieve (3), i.e., no settings file at all, a user should be able to:

- Create a brand new python env
- Run `pip install flowcept`
- Start up the database services using docker compose
- Run the basic script on the README file: https://github.com/ORNL/flowcept?tab=readme-ov-file#simple-example-with-decorators-instrumentation

Currently, it is mandatory that the user specifies the `FLOWCEPT_SETTINGS_PATH` env var. I would like to modify this behavior. 

This is the behavior I propose, as I believe it will work well given the latest experience with Flowcept on a cluster and daily development. I am open to discussion.

- If `FLOWCEPT_SETTINGS_PATH` is not defined, the configs.py will look to see if there is a valid file in `~/.flowcept/settings.yaml`.
     -  If there is, the variables in the configs.py should be updated according to what is in this yaml file. 
     -  If there isn't, it's ok. The variables in the configs.py should be updated with the default values. 
- If the env var is defined but the file it points to does not exist, an exception should be thrown.
- If the env var is defined with a valid yaml file, the variables in the configs.py should be updated according to what is the provided yaml file.

**Comment by renan-souza** [comment]
Linked issue: https://github.com/ORNL/flowcept/issues/119

**Comment by renan-souza** [comment]
Let's try to keep source code and configs as separated as we can. Let's avoid having hardcoded raw YAML or even a configs dict hardcoded in the Python source configs.py file. We can have the default settings specified in their own configs file.

**Comment by wigging** [comment]
> Start up the database services using docker compose

So the database services (like mongodb) are required to run Flowcept?

**Comment by renan-souza** [comment]
> > Start up the database services using docker compose
> 
> So the database services (like mongodb) are required to run Flowcept?

Yes. The current implementation of Flowcept requires a database (currently, MongoDB) and a MQ (currently, Redis or Kafka).

**Comment by wigging** [comment]
The README states that MongoDB and Redis services are needed to take full advantage of flowcept. So this makes me think that MongoDB and Redis are optional. The simple example in the README doesn't require any of these services or does it?

**Comment by renan-souza** [comment]
> The README states that MongoDB and Redis services are needed to take full advantage of flowcept. So this makes me think that MongoDB and Redis are optional. The simple example in the README doesn't require any of these services or does it?

This is outdated. Thanks for pointing this out. I will fix it in the README.

**Comment by renan-souza** [comment]
TL;DR:

It remained as it was, except that we are now including the settings file into the package directly, eliminating the extra step to create the settings file.

Longer explanation:

The settings YAML file will remain necessary because the adapters rely on it to determine their default values. 

Flowcept works both as a package and a complete system with backend dependencies, and when adapters are enabled, it relies on additional services like MLFlow and Dask. To operate in this complex setup, the settings are needed, particularly for integration with these external systems.

In the future, we could consider splitting settings into multiple YAML files—one per adapter—for modularity, but this would require a longer implementation time.

In the end, I included sample_settings.yaml directly in the package as a resource (which was actually the intention in the original design) by modifying pyproject.toml. This allows for a simple run configuration without forcing users to install unnecessary dependencies or specify a separate YAML file through an environment variable to test the simple case.

I also modified the flowcept/__init__.py 
The current behavior is as follows:

The settings in settings.yaml are overridden by any file specified in the environment variable FLOWCEPT_SETTINGS_PATH.
If no path is set, Flowcept will check for a file at ~/.flowcept/settings.yaml. If found, it will use this file.
If neither option is available, Flowcept will fall back to sample_settings.yaml from the package’s resources directory, which contains all default values.


This closes the issue https://github.com/ORNL/flowcept/issues/119

**Comment by renan-souza** [comment]
This [PR](https://github.com/ORNL/flowcept/pull/157) closes this issue as well. I opened a new one for the CLI implementation.
